extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java

```diff
@@ -89,10 +93,14 @@ public BeanArchiveIndexBuildItem build(LiveReloadBuildItem liveReloadBuildItem)
             liveReloadBuildItem.setContextObject(BeanArchives.PersistentClassIndex.class, index);
         }
 
+        IndexView additionalBeanIndex = additionalBeanIndexer.complete();
```

Comment:
Well, this index contains not only the generated beans but also beans added via ``AdditionalBeanBuildItem``s. I'm not quite sure whether it's OK. We need someone who knows more about the `ReflectiveHierarchyStep`.  @gsmet @geoand WDYT?

Reply:
It wouldn't be a problem for the `ReflectiveHierarchyStep`, but it could potentially cause problem in other places if people start using this build item.

Reply:
so what would be an alternative to get only beans generated? I can change this to whatever ... just need to know what this whatever is :)

Reply:
@geoand Could you please elaborate what issues this might cause? I mean it has classes indexed that are part of the application so should not cause any side effects ... same as if people would use the ‘BeanArchiveIndexBuildItem’

Reply:
@mkouba @geoand took another try to use only https://github.com/quarkusio/quarkus/blob/3e3aa03fca248b3639365f2f9e042ab0f728130a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java#L81 in new indexer but that proved to be not enough ... not all classes where found by `ReflectiveHierarchyStep`

any other suggestions?

Reply:
It might cause problems because people thinking that the index will contain only generated classes, well end up seeing additional classes as well.

In the case of reflective hierarchy step it won't be a problem, but in general it's not safe to provide an index with more classes than expected.

I'll check your latest modification tomorrow

Reply:
To be honest, I am not sure what the proper solution would be in this case...

Reply:
@geoand @mkouba so I took another approach and not `GeneratedClassesIndexBuildItem` is produced from the extension that generates these classes. That is (as @mkouba suggested) because `GeneratedClassesIndexBuildItem` is `MultiBeanItem` each extension can produce its own index with generated classes which makes it way better and actually serves the purpose of this new BuildItem. Wdyt?

Reply:
That sounds like a reasonable compromise to me. But I think the naming of the build item could be better :)

Reply:
@geoand any suggestion for better name of the build item?

Reply:
I'm notoriously terrible at naming :P. 

The name of the build item sounds too generic to me TBH and I am afraid that others might try to use it for other purposes. Perhaps just some strongly worded Javadoc will do the trick.
I'll let @mkouba decide since he is way better at naming that me :)

Reply:
@geoand that makes two of us :)

current name was actually suggested by @mkouba though if better name is found let's change it. Meantime I will add some javadocs to explain the purpose of this build item.

Reply:
:+1:

Reply:
@geoand javadoc updated, is it any better?

Reply:
LGTM!

Reply:
let's wait for @mkouba if he has any better name for the build item.